nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -110,6 +110,24 @@ impl<'a> Planner<'a> {
                 SledState::Active => {}
             }
 
+            // Remove any zones which rely on expunged disks.
+            if self
+                .blueprint
+                .sled_ensure_zones_not_using_expunged_disks(*sled_id)?
+                == Ensure::Added
+            {
+                info!(
+                    &self.log,
+                    "found sled with expunged disks (will remove the zones using them)";
+                    "sled_id" => %sled_id
+                );
+                self.blueprint.comment(&format!(
+                    "sled {}: remove expunged disks",
+                    sled_id
+                ));
+                continue;
```

Comment:
This seems to match the pattern below of "if we have any changes to make, do them and stop making further changes".

It seems plausible that we could remove an expunged zone and also do other allocations, but I'm not sure what our motivation is for "updating the blueprints via small steps", and wasn't confident enough to break that mold.